### PR TITLE
Add metadataWhitelist option to minimize topic metadata get

### PR DESCRIPTION
**Pull request by mmchen**
If metadataWhitelist is passed (null or [] accepted), it will prune metadata get request to only whitelist topic list.

### CODE REVIEW

File: lib/kafka_base_producer.js [hunk 1]
```diff
@@ -85,6 +85,14 @@ function KafkaBaseProducer(options, producerType) { // eslint-disable-line
         } else {
             self.batchingBlacklist = options.batchingBlacklist || [];
         }
+        self.metadataWhitelist = null;
+        if ('metadataWhitelist' in options) {
+            // expecting a list of strings, either [] or null should also be ok
+            self.metadataWhitelist = [];
+            if (options.metadataWhitelist === null || Array.isArray(options.metadataWhitelist)) {
```
Comment (cloudXane) [inline lib/kafka_base_producer.js]: if isArray(null) returns false, we might simplify this if-check

Comment (mmchen) [inline lib/kafka_base_producer.js]: Done

**Comment by xhl1988** [comment]
lgtm

**Comment by proflayton** [comment]
lgtm! Thanks for making this change :)

**Pull request merged**